pkg/plugin/manager.go

```diff
@@ -65,8 +66,8 @@ func (m *Manager) DownloadProviders(ctx context.Context, providers []registry.Pr
 	log.Debug().Interface("providers", providers).Msg("Downloading required providers")
 	downloaded := make([]registry.ProviderBinary, len(providers))
 	for i, rp := range providers {
-		if _, ok := m.clients[rp.String()]; ok {
-			log.Debug().Str("name", rp.Name).Str("version", rp.Version).Msg("Skipping provider download, using reattach instead")
+		if _, ok := m.clients[rp.Name]; ok {
```

Comment:
`clients` are de-serialized from the `.cq_reattach` file which includes only the name of the provider. `rp.String()` format is `name@version`, hence this was failing to match the providers.